### PR TITLE
add elem in map

### LINKED ISSUE #118: Map contains check (elem in map) implementation
We have `in` test after 64bf276 commit (ie. `elem in container`) however we currently only support lists, The map implementation would be fairly simple.

## How to implement
- You can find the TODO [here](https://github.com/ThakeeNathees/pocketlang/blob/64bf27674a068a008a78a545e7e353d5659dd1e9/src/pk_core.c#L1305)
- Use the [`mapGet()`](https://github.com/ThakeeNathees/pocketlang/blob/64bf27674a068a008a78a545e7e353d5659dd1e9/src/pk_var.h#L533-L535) function to check if the element exists.
- Return true if it exists.

## Finally

Don't forgot to write some tests at `tests/lang/basics.pk`
```ruby
assert('key' in {'key':'value'})
assert(!('foo' in {'bar':'baz'}))
```

### PR DESCRIPTION
**Pull request by tsujp**
*Edit Fix: #118*

### CODE REVIEW

File: src/pk_core.c [hunk 1]
```diff
@@ -1302,7 +1302,13 @@ bool varContains(PKVM* vm, Var elem, Var container) {
       return false;
     } break;
 
-    case OBJ_MAP:
+    case OBJ_MAP: {
+      Map* map = (Map*)AS_OBJ(container);
+      return mapGet(map, elem) == VAR_UNDEFINED
+        ? false
+        : true;
```
Comment (ThakeeNathees) [inline src/pk_core.c]: This can be simplified with  `return !IS_UNDEF(mapGet(map, elem));`

**Review by ThakeeNathees — CHANGES_REQUESTED** [review]
Thank you for taking your time and effort to make this PR,

Just a tiny change I'd like before merging it.

File: src/pk_core.c (hunk 1 above)
Comment (tsujp) [reply src/pk_core.c]: Neat!

**Comment by tsujp** [comment]
I've made the requested change @ThakeeNathees -- no worries you basically spelled it out in the issue after all :)

**Review by ThakeeNathees — APPROVED** [review]
Thanks!

We're glad to have you here 🎉

**Pull request merged**